include/boost/thread/pthread/condition_variable.hpp

```diff
@@ -155,6 +169,14 @@ namespace boost
         pthread_mutex_t internal_mutex;
         pthread_cond_t cond;
 
+#ifdef BOOST_THREAD_USES_CHRONO
+#ifdef BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC
```

Comment:
This type should be declared only once. What about moving it to a specific file?

Reply:
That's a good idea.

Reply:
What about thread/detail/internal_clock.hpp?